File: project/dahee-onboarding/survey-core/src/main/java/com/survey/core/repository/SurveyItemRepository.java
package com.survey.core.repository;

import com.survey.core.entity.SurveyItem;
import org.springframework.data.jpa.repository.JpaRepository;

import java.util.List;

public interface SurveyItemRepository extends JpaRepository<SurveyItem, Long> {
    List<SurveyItem> findBySurveyId(Long surveyId);

    List<SurveyItem> findBySurveyIdAndIsDeletedFalse(Long id);

}

File: project/sungmin-onboarding/survey-service/src/main/kotlin/com/okdori/surveyservice/repository/SurveyItemRepository.kt
package com.okdori.surveyservice.repository

import com.okdori.surveyservice.domain.Survey
import com.okdori.surveyservice.domain.SurveyItem
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.data.jpa.repository.Modifying
import org.springframework.data.jpa.repository.Query

/**
 * packageName    : com.okdori.surveyservice.repository
 * fileName       : SurveyItemRepository
 * author         : okdori
 * date           : 2025. 6. 16.
 * description    :
 */
interface SurveyItemRepository: JpaRepository<SurveyItem, String> {
    fun findBySurveyAndIsDeleted(survey: Survey, isDeleted: Boolean): List<SurveyItem>

    @Modifying
    @Query("UPDATE SurveyItem si SET si.isDeleted = true WHERE si.survey = :survey")
    fun softDeletedBySurvey(survey: Survey)
}
